### PR TITLE
FFA-27  Connect google button with firebase auth and create login/register

### PR DESCRIPTION
**Pull request by Coshiloco**
- Add Login with Email and Password
- Add Google Sign In Button
- Add register 
- Add logout
- Create a fake Home and exit when back is pressed

### CODE REVIEW

File: app/src/main/java/com/digitalsolution/familyfilmapp/di/GoogleSignInModule.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
```
Comment (rndevelo) [inline app/src/main/java/com/digitalsolution/familyfilmapp/di/GoogleSignInModule.kt:14]: I think that this module should has more things (?)

File: app/src/main/java/com/digitalsolution/familyfilmapp/repositories/LoginRepositoryImpl.kt [hunk 2]
```diff
@@ -37,9 +38,24 @@ class LoginRepositoryImpl @Inject constructor(
         awaitClose()
     }
 
+    override fun loginWithGoogle(idToken: String): Flow<Result<AuthResult>> = callbackFlow {
```
Comment (rndevelo) [inline app/src/main/java/com/digitalsolution/familyfilmapp/repositories/LoginRepositoryImpl.kt:41]: When I logged with google with app, onSuccess crashed, we should look good this

File: app/src/main/java/com/digitalsolution/familyfilmapp/ui/screens/login/LoginScreen.kt [hunk 3]
```diff
@@ -36,13 +44,16 @@ fun LoginScreen() {
     var textPasswordState by remember { mutableStateOf("") }
     val snackBarHostState = remember { SnackbarHostState() }
 
+    val loginViewModel: LoginViewModel = hiltViewModel()
+
     Scaffold(snackbarHost = { SnackbarHost(snackBarHostState) }) { innerPadding ->
         LoginContent(
```
Comment (rndevelo) [inline app/src/main/java/com/digitalsolution/familyfilmapp/ui/screens/login/LoginScreen.kt:50]: We should create a new branch to create dataclass for parameters

File: app/src/main/java/com/digitalsolution/familyfilmapp/ui/screens/login/LoginViewModel.kt [hunk 4]
```diff
@@ -32,4 +39,24 @@ class LoginViewModel @Inject constructor(
         // TODO
     }
 
+    fun googleSignIn(): Intent {
```
Comment (rndevelo) [inline app/src/main/java/com/digitalsolution/familyfilmapp/ui/screens/login/LoginViewModel.kt:42]: We can call this without return function (?)

**Review by rndevelo — CHANGES_REQUESTED** [review]
Look comments Pablo, we need look this

File: app/src/main/java/com/digitalsolution/familyfilmapp/ui/screens/login/LoginViewModel.kt (hunk 4 above)
Comment (hgarciaalberto) [reply app/src/main/java/com/digitalsolution/familyfilmapp/ui/screens/login/LoginViewModel.kt:42]: I've already told him the same and I think it is already fixed

**Review by hgarciaalberto — APPROVED** [review]
Everything's fine

**Review by Coshiloco — COMMENTED** [review]
Awesome Features

**Review by rndevelo — APPROVED** [review]

**Review by rndevelo — APPROVED** [review]

**Review by rndevelo — APPROVED** [review]

**Pull request merged**